enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -2099,10 +2099,10 @@ GradientUtils *GradientUtils::CreateFromClone(
 }
 
 DiffeGradientUtils *DiffeGradientUtils::CreateFromClone(
-    EnzymeLogic &Logic, bool topLevel, Function *todiff, TargetLibraryInfo &TLI,
-    TypeAnalysis &TA, DIFFE_TYPE retType, bool diffeReturnArg,
-    const std::vector<DIFFE_TYPE> &constant_args, ReturnType returnValue,
-    Type *additionalArg) {
+    EnzymeLogic &Logic, DerivativeMode mode, Function *todiff,
+    TargetLibraryInfo &TLI, TypeAnalysis &TA, DIFFE_TYPE retType,
+    bool diffeReturnArg, const std::vector<DIFFE_TYPE> &constant_args,
+    ReturnType returnValue, Type *additionalArg) {
```

Comment:
Add an assert that this is ReverseModeGradient or ReverseModeCombined (and also right now potentially forwardMode, but I think that would be better abstracted into a different subclass of gradientutils).

Reply:
done